### PR TITLE
Edit items 1

### PR DESCRIPTION
**Pull request by EthanJStark**

### CODE REVIEW

**Comment by EthanJStark** [comment]
@jrob8577  Seeking an in-progress code review, attempting to do so according to what you spoke of in stand-up.  My main interest is the manner in which we are structuring the tree of item > individual > title etc.  Thanks!

File: public/javascripts/items.js [hunk 1]
```diff
@@ -11,24 +11,50 @@ const params = completed =>
   Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
 
 $(document).ready( () => {
-  $( '.title' ).click( event => {
+  $( '.glyphicon' ).click( event => {
```
Comment (jrobcodes) [inline public/javascripts/items.js]: `glyphicon` is not an intention revealing name - it names what we're _using_, not what it _is_

File: public/javascripts/items.js [hunk 2]
```diff
@@ -11,24 +11,50 @@ const params = completed =>
...
+      if( json.success ) {
+        element.data( 'completed', completed )
 
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
+        const parent = $( `.individual[data-id=${id}]` )
```
Comment (jrobcodes) [inline public/javascripts/items.js]: I don't know if I like this, but here's a thing:

``` es6
$( `.individual[data-id=${id}]` )[ completed ? 'addClass' : 'removeClass' ]( 'completed' )
```

File: public/javascripts/items.js [hunk 3]
```diff
@@ -11,24 +11,50 @@ const params = completed =>
...
-      })
+      }
+    })
   })
 })
+
+const paramsTwo = (title, description) =>
+  Object.assign( {}, fetchParams, { body: JSON.stringify({ title, description }) })
+//
+// $(document).ready( () => {
```
Comment (jrobcodes) [inline public/javascripts/items.js]: 😮  comments

File: public/javascripts/items.js [hunk 4]
```diff
@@ -11,24 +11,50 @@ const params = completed =>
...
     const id = element.data( 'id' )
     const completed = ! element.data( 'completed' )
 
     fetch( `/items/${id}/completed`, params( completed ))
-      .then( result => result.json() )
-      .then( json => {
-        if( json.success ) {
-          element.data( 'completed', completed )
+    .then( result => result.json() )
+    .then( json => {
```
Comment (jrobcodes) [inline public/javascripts/items.js:79]: Would love to explore ways to condense this a little bit, or maybe extract the responsibility of processing the response into its own function:

``` es6
const updateElement = element => result => {
  // Let's talk about a way to remove this - we can respond with a non-OK response code, then 
  // handle this with a catch instead... 
  if( result.success ) {
    // Do the item processing
  }
}
```

File: views/items/index.pug [hunk 5]
```diff
@@ -2,8 +2,10 @@ extends ../layouts/authenticated-layout
 
 mixin display( nodes, depth )
   each entry in nodes
-    .item(data-id=entry.id, class={completed: entry.completed})
-      .title(data-id=entry.id, data-completed=entry.completed)= entry.title
+    .item(data-id=entry.id)
+      form.individual(data-id=entry.id, class={completed: entry.completed}, data-completed=entry.completed, action=`/items/${entry.id}/completed`, method='post')
```
Comment (jrobcodes) [inline views/items/index.pug]: Would like to improve our naming conventions here so that the items are named such that they are meaningful in our domain.

**Review by jrobcodes — CHANGES_REQUESTED** [review]
Left some feedback, we can discuss when you have time.  :)

**Pull request closed without merging**